### PR TITLE
Deprecate requireAnyRole and requireAllRoles

### PR DESCRIPTION
**Pull request by leleuj**

### CODE REVIEW

File: pac4j-core/src/main/java/org/pac4j/core/authorization/RequireAllRolesAuthorizer.java [hunk 1]
```diff
@@ -27,28 +29,46 @@
  */
 public class RequireAllRolesAuthorizer<U extends UserProfile> implements Authorizer<U> {
 
-    private final String[] expectedRoles;
+    private String[] roles;
```
Comment (mmoayyed) [inline pac4j-core/src/main/java/org/pac4j/core/authorization/RequireAllRolesAuthorizer.java]: Why not use a List?

Comment (mmoayyed) [inline pac4j-core/src/main/java/org/pac4j/core/authorization/RequireAllRolesAuthorizer.java]: Or better yet, a Set?

File: pac4j-core/src/main/java/org/pac4j/core/authorization/RequireAnyRoleAuthorizer.java [hunk 2]
```diff
@@ -27,28 +29,46 @@
  */
 public class RequireAnyRoleAuthorizer<U extends UserProfile> implements Authorizer<U> {
 
-    private final String[] expectedRoles;
+    private String[] roles;
```
Comment (mmoayyed) [inline pac4j-core/src/main/java/org/pac4j/core/authorization/RequireAnyRoleAuthorizer.java]: See my comment above please

File: pac4j-core/src/main/java/org/pac4j/core/authorization/RequireAllRolesAuthorizer.java (hunk 1 above)
Comment (leleuj) [inline pac4j-core/src/main/java/org/pac4j/core/authorization/RequireAllRolesAuthorizer.java]: No specific reason. A `Set` would be better indeed.

**Comment by leleuj** [comment]
Updated according to @mmoayyed 's feedbacks.

**Comment by mmoayyed** [comment]
:+1:

**Pull request merged**